src/node_file.cc

```diff
@@ -1853,6 +1880,21 @@ static void Open(const FunctionCallbackInfo<Value>& args) {
   CHECK(args[2]->IsInt32());
   const int mode = args[2].As<Int32>()->Value();
 
+  // Open can be called either in write or read
+  if (flags == O_RDWR) {
+    // TODO(rafaelgss): it can be optimized to void n*m
+    THROW_IF_INSUFFICIENT_PERMISSIONS(
+        env, permission::PermissionScope::kFileSystemIn, *path);
+    THROW_IF_INSUFFICIENT_PERMISSIONS(
+        env, permission::PermissionScope::kFileSystemOut, *path);
+  } else if ((flags & ~(O_RDONLY | O_SYNC)) == 0) {
```

Comment:
That's not cross-platform (windows build failing). I'll figure out a better approach.